### ISSUE TITLE
Minify failed when assignment statement in closure

### ISSUE THREAD

**Issue by JSerFeng** [swc-project/swc#7657]
### Describe the bug

The whole variable `foo` is gone after compress. And if I set isModule: true, output will become empty.

### Input code

```typescript
let foo;
(function () {
  foo = 1;
})();
```

### Config

```json
{
  "jsc": {
    "target": "es2020",
    "parser": {
      "syntax": "typescript",
    },
    "minify": {
      "mangle": false,
      "compress": {
        "arguments": false,
        "arrows": true,
        "booleans": true,
        "booleans_as_integers": false,
        "collapse_vars": true,
        "comparisons": true,
        "computed_props": true,
        "conditionals": true,
        "dead_code": true,
        "directives": true,
        "drop_console": false,
        "drop_debugger": true,
        "evaluate": true,
        "expression": false,
        "hoist_funs": false,
        "hoist_props": true,
        "hoist_vars": false,
        "if_return": true,
        "join_vars": true,
        "keep_classnames": false,
        "keep_fargs": true,
        "keep_fnames": false,
        "keep_infinity": false,
        "loops": true,
        "negate_iife": true,
        "properties": true,
        "reduce_funcs": false,
        "reduce_vars": false,
        "side_effects": true,
        "switches": true,
        "typeofs": true,
        "unsafe": false,
        "unsafe_arrows": false,
        "unsafe_comps": false,
        "unsafe_Function": false,
        "unsafe_math": false,
        "unsafe_symbols": false,
        "unsafe_methods": false,
        "unsafe_proto": false,
        "unsafe_regexp": false,
        "unsafe_undefined": false,
        "unused": true,
        "const_to_let": true,
        "pristine_globals": true
      }
    },
  },

  "minify": true,
  "isModule": false,
  "module": {
    "type": "commonjs"
  },
}
```

### Playground link

https://play.swc.rs/?version=1.3.66&code=H4sIAAAAAAAAA8tJLVFIy8%2B35tJIK81LLsnMz1PQ0FSo5lIAiSrYKhhac9VqamhaAwBMhrwyKAAAAA%3D%3D&config=H4sIAAAAAAAAA32US27jMAyG9z1F4HUXRZc9wOzmDIJiUY4ykmiIVBqjyN2HfrVpTGdj2Pz4kH9S%2FHo5HJoztc3H4Ute5YNt6YDluwF6f3t%2Fa15ne28LQfn2EwsNme119OShB2pL6Hnxvi1BKeTgh%2FugZHMXQSzeRoLX1dxi6gsQ3bmKVY5SE2SmR%2F%2BJFfwcAZd6bz8iRrD5CTGWTMgMHRQtcYsx2p7AXGxRsowntSUQaiVGWBmc6Qv2Ks8ucMAsNbfUgXWmRQcKCgVaDhfQwqSWhGXCrbArdnCsXTf17yEaLjZWy0pNuE4tkdMqWU8YiI2vWZNwhjsazHAR9zEyeFOAa8nbuDOGvNOTfwCiQLRE2SbQ8k4eXuZpL9o%2FjQzZyyTzoPCI6l9m6ERUE4JXlB2VgcJB62YBV1sYlW214yx4Rz4KDgx4L7OipKbPwO1JKzpeYPQKkP5ar03VDMz3Ldzh44V4gv%2FIX7I%2BYItHsnzapzSkI8YnBRLwCd0TB2kF4z4usiWu%2FT6v2YGMBjjVpdIEtktALgCjidOe3cyGXA%2FJaLqIx581sTjcftarPF7uV%2Byapwn0F139vQuatJrWNS8dH1e3NChhPlMzZ7z9B%2Fm7PlURBgAA

### Expected behavior

`foo` should be assigned

### Actual behavior

Assignment is gone

### Version

1.3.66

**Comment by JSerFeng** [comment]
Find out it's the `unused` configurations caused this, I'm trying to dig deeper

**Comment by JSerFeng** [comment]
Not sure if this is fixed, can't produce in the SWC compress test

**Comment by kdy1** [comment]
@JSerFeng It may have different value for `isModule` or something. Can you try creating a test in https://github.com/swc-project/swc/tree/7bf4ab39b0e49759d9f5c8d7f989b3ed010d81a7/crates/swc_ecma_minifier/tests/fixture/issues ?

**Comment by Austaras** [comment]
This issue has already been fixed in #7568 

@swc/wasm and @swc/wasm-web has not been released since 1.3.66. This should be addressed soon as possible

**Comment by kdy1** [comment]
I'll work on https://github.com/swc-project/swc/issues/7663 right away. @Austaras  Thank you!
